specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
+specific congestion control algorithm or parameters.
+[Brokers](./overview.md#broker) and [Channels](./overview.md#channel) MUST
+implement congestion control and MUST implement retries.
 
 ### Observability
 
-CloudEvents received by Sink MAY have
-[Distributed Tracing Extension Attribute](https://github.com/cloudevents/spec/blob/v1.0/extensions/distributed-tracing.md).
-
-### Event reply contract
-
-An event sender supporting event replies SHOULD include a `Prefer: reply` header
-in delivery requests to indicate to the sink that event reply is supported. An
-event sender MAY ignore an event reply in the delivery response if the
-`Prefer: reply` header was not included in the delivery request.
-
-An example is that a Broker supporting event reply sends events with an
-additional header `Prefer: reply` so that the sink connected to the Broker knows
-event replies will be accepted. While a source sends events without the header,
-in which case the sink may assume that any event reply will be dropped without
-error or retry attempt. If a sink wishes to ensure the reply events will be
-delivered, it can check for the existence of the `Prefer: reply` header in the
-delivery request and respond with an error code if the header is not present.
-
-### Data plane contract for Sources
-
-See [Source Delivery specification](sources.md#source-event-delivery)
-for details.
-
-### Data plane contract for Channels
-
-See [Channel Delivery specification](channel.md#data-plane) for details.
-
-### Data plane contract for Brokers
-
-See [Broker Delivery specification](broker.md)
-
-## Changelog
-
-- 2020-04-20: `0.13.x release`: initial version that documents common contract
-  for sinks, sources, channels and brokers.
+Event senders MAY add or update CloudEvents attributes before sending to
+implement observability features such as tracing; in particular, the
+`traceparent` and `tracestate` distributed tracing attributes defined by
+[W3C](https://www.w3.org/TR/trace-context/) and
+[CloudEvents](https://github.com/cloudevents/spec/blob/v1.0/extensions/distributed-tracing.md)
+may be modified in this way for each delivery attempt of the same event.
+
+This specification does not mandate any particular logging or metrics
+aggregation, nor a method of exposing observability information to users
+configuring the resources. Platform administrators SHOULD expose event-delivery
+telemetry to users through platform-specific interfaces, but such interfaces are
+beyond the scope of this document.
+
+<!-- TODO: should we mention RECOMMENDED spans or RECOMMENDED metrics like in
+https://github.com/knative/specs/blob/main/specs/eventing/channel.md#observability?
+-->
+
+### Derived (Reply) Events
+
+In some applications, an event receiver might emit an event in reaction to a
+received event. Components MAY to choose to support this pattern by accepting an
+encoded CloudEvent in the HTTP response. The sender SHOULD NOT assume that a
```

Comment:
the SHOULD NOT here is very interesting when paired with the "might emit an event in reaction to a received event". The sentence above sure implies a relationship to me. So, if we're explicitly saying that it could be a totally unrelated event how can components like a Sequence provide any guarantees that it'll work properly?  Can you explain why this SHOULD NOT is needed? Seems like we're opening the door for people to ask "ok, if they can be unrelated then how can I indicate that it IS related? Or how do I send a related response later?"

Reply:
@n3wscott @slinkydeveloper @vaikas do you know why this is here? W/o a clear reason for why it could be a random event, I'd prefer to remove this and then leave the implication be that it's related.  That leave it open for us to "clarify" it later by adding text to say that it can be random or mandate that it be related once we get more feedback.

Reply:
:+1: with removing it. I don't know why that sentence is here.